### ISSUE TITLE
Silent mode signal mapping

### ISSUE THREAD

**Issue by UlfHRO** [tp1de/ioBroker.ems-esp#159]
**Describe the bug**  
The value of"silentmode" is showing wrong status indication for status "on(2)". If "on(2)" is written by ioBroker (or locally at the heatpump) the status feedback in iobroker is "auto(1)", even the function is set correctly to "on(2)"...
Status feedback for "auto(1)" is correct, means for both "auto" and "on" the same status "auto" is indicated in ioBroker. In the EMS-ESP all is correct, commands and feedback work as expected. So, the failure seems to be in the ioBroker adapter...

** Releated to gateway(s) and/or adapter configuration

B Which version of the ems-esp adapter are you using? latest 5.0.6
C is this bug related to the ems-esp or km200 gateway? (which gateway hardware are you using?) ESP32-D0WD-V3
D please descripe software versions of the ems-esp firmware and/or km200 gateway: 3.7.1
E please post screenshot(s) of the adapter config

**Expected behavior**  
Please correct the signal mapping for "silent mode"
- 0 is off
- 1 is auto
- 2 is on

**Additional context**

**Comment by tp1de** [comment]
I do not remap anything within the adapter.
Please post the object data for this field within object browser.

**Comment by UlfHRO** [comment]
Hi Thomas,
I tested again and the issue happens every time... Only in ioBroker, all correct in EMS-ESP...
Please find below the requested object data...
Thanks...

{
  "_id": "ems-esp.0.boiler.silentmode",
  "type": "state",
  "common": {
    "id": "boiler.silentmode",
    "name": "ems: silent mode",
    "type": "mixed",
    "read": true,
    "write": true,
    "role": "level",
    "states": {
      "0": "off",
      "1": "auto",
      "2": "on"
    },
    "custom": {
      "influxdb.0": {
        "enabled": true,
        "storageType": "",
        "aliasId": "",
        "debounceTime": 0,
        "blockTime": 0,
        "changesOnly": false,
        "changesRelogInterval": 0,
        "changesMinDelta": 0,
        "ignoreBelowNumber": "",
        "disableSkippedValueLogging": false,
        "enableDebugLogs": false,
        "debounce": 0
      }
    }
  },
  "native": {
    "ems_enum": [
      "off",
      "auto",
      "on"
    ],
    "ems_type": "enum",
    "visible": true,
    "ems_command": "silentmode",
    "ems_device": "boiler",
    "ems_id": "",
    "ems_api": "3.7.1"
  },
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  },
  "from": "system.adapter.admin.0",
  "user": "system.user.admin",
  "ts": 1735152646201
}


![Bildschirmfoto 2024-12-25 um 19 56 46](https://github.com/user-attachments/assets/24df13bf-c8f8-48d1-92db-d0660e7f2915)

**Comment by tp1de** [comment]
It is difficult to understand your problem.
The actual values are taken from the api call:  `ems-esp/api/boiler`

Please check the value returned

**Comment by UlfHRO** [comment]
Hi Thomas,

Yes, the command is correctly executed, but displayed value after few seconds is wrongly changing to "auto"...

The api get is correct for status "on". Still ioBroker is changing to "auto" (1) after a few seconds...
The issue is within the ioBroker adapter... Sorry...
Please see below the api feedback:

{"name":"silentmode","fullname":"silent mode","circuit":"","index":2,"enum":["off","auto","on"],"value":"on","type":"enum","readable":true,"writeable":true,"visible":true}

Thanks for your support!

**Comment by tp1de** [comment]
Please check the value from .../api/boiler

**Comment by UlfHRO** [comment]
Same there: correct status shown as "on", but iobroker shows "auto"...

"silentmode":"on"

**Comment by tp1de** [comment]
Hmmm I do not understand why this value is not used.
Do you have this issue with other enum objects?

**Comment by UlfHRO** [comment]
No, not at all... I read a lot values, all consistent and ok... Only this is changing after a few seconds wrongly to "auto"...

**Comment by tp1de** [comment]
If you change to "auto" does it then shows "off"? (allways one lower?)

**Comment by UlfHRO** [comment]
No, off and auto are ok... Only "on" gives the wrong value...

**Comment by tp1de** [comment]
I do not have any idea what is happening. 
The code for enums is more then 2 years old and hasn't changed.
I can not test myself. I will review again ..

**Comment by UlfHRO** [comment]
Other enums are ok... Only this is wrong... Thanks...

**Comment by tp1de** [comment]
I will prepare an adapter version to be installed from Github selecting boiler.silentmode to write the read values into log.
You are not using the km200 structure isn't it? (obect is boiler.silentmode)

**Comment by UlfHRO** [comment]
Hi Thomas,

**Auto:**
ems-esp.0
2024-12-26 17:36:09.753	info	boiler.silentmode index 1

ems-esp.0
2024-12-26 17:36:09.731	info	boiler.silentmode auto

**Off:**
ems-esp.0
2024-12-26 17:36:39.765	info	boiler.silentmode index 0

ems-esp.0
2024-12-26 17:36:39.745	info	boiler.silentmode 0

**On:**
ems-esp.0
2024-12-26 17:37:09.746	info	boiler.silentmode index 1

ems-esp.0
2024-12-26 17:37:09.732	info	boiler.silentmode 1

**Comment by UlfHRO** [comment]
yes...

**Comment by tp1de** [comment]
But the you should get on every read two lines:

1st with value "off", "auto" or "on" and then 2nd the index 0,1 or 2

**Comment by UlfHRO** [comment]
I get two lines each, but with the content as shared above...

**Comment by tp1de** [comment]
Please change ems-esp settings to index and test again

**Comment by UlfHRO** [comment]
Then it works... But what will happen to all my other enums with this changed setting...?

**Auto:**
ems-esp.0
2024-12-26 18:10:25.246	info	boiler.silentmode index 1

ems-esp.0
2024-12-26 18:10:25.232	info	boiler.silentmode 1

**Off:**
ems-esp.0
2024-12-26 18:10:55.174	info	boiler.silentmode index 0

ems-esp.0
2024-12-26 18:10:55.156	info	boiler.silentmode 0

On:
ems-esp.0
2024-12-26 18:12:25.194	info	boiler.silentmode index 2

ems-esp.0
2024-12-26 18:12:25.176	info	boiler.silentmode 2

**Comment by tp1de** [comment]
> Then it works... But what will happen to all my other enums with this changed setting...?

It should work both ways. The API for silentmode seems to have errors on mode "value".
Expected values are off,auto, on and not a mix of index and letters.
 
Have you tested the last dev version?
I

**Comment by UlfHRO** [comment]
No, I have not "tested" the last dev version... Nothing in the change log for this topic... I use latest 3.7.1.
Shall I give it a try or better assign the problem to the EMS-ESP team...?

**Comment by tp1de** [comment]
Please test the latest dev with "value". 
f the error is still there I will open a GH issue for ems-esp while referencing this iobroker GH issue.

**Comment by UlfHRO** [comment]
Same result with latest dev:

ems-esp.0
2024-12-26 18:43:14.493	info	boiler.silentmode index 1

ems-esp.0
2024-12-26 18:43:14.475	info	boiler.silentmode 1

ems-esp.0
2024-12-26 18:42:59.520	info	boiler.silentmode index 1

ems-esp.0
2024-12-26 18:42:59.499	info	boiler.silentmode auto

ems-esp.0
2024-12-26 18:42:44.631	info	boiler.silentmode index 0

ems-esp.0
2024-12-26 18:42:44.498	info	boiler.silentmode 0

**Comment by UlfHRO** [comment]
I have now changed back your adapter to the official latest 5.1.0 and changed the EMS-ESP setting for Enum to index until the bug is fixed by the EMS-ESP-Team...

Thanks a lot Thomas and have a good Xmas time...

**Comment by tp1de** [comment]
https://github.com/emsesp/EMS-ESP32/issues/2306

> Thanks a lot Thomas and have a good Xmas time...

Have a good time too

**Comment by tp1de** [comment]
@UlfHRO 
I uploaded a changed adapter version to Github. 
Could you please check if "Value" is now working?

**Comment by UlfHRO** [comment]
Hi Thomas,
Yes, confirmed... Works as expected with setting "value" for Enum...
Thanks a lot and best regards,
Ulf

**Comment by tp1de** [comment]
Fixed with 5.1.2

**Comment by UlfHRO** [comment]
Thanks Thomas!
